07-complex-data-type/src/ComplexDataTypeSample.java

```diff
@@ -0,0 +1,258 @@
+import com.terracottatech.store.*;
+import com.terracottatech.store.definition.ComplexDataParseException;
+import com.terracottatech.store.definition.ComplexDataParser;
+
+import java.util.*;
+
+import static com.terracottatech.store.TypedValue.typedValue;
+
+public class ComplexDataTypeSample {
```

Comment:
It would be nice to show some actual interaction between and use of complex data types stored in a Dataset. That way you could use the functional builders in the way they are intended by filtering and mapping the stream of records.